### PR TITLE
add bar indicator

### LINKED ISSUE #608: Alarms: Ack, Open, Close buttons don’t work
# Type of issue

- [x ] Bug
- [ ] New feature
- [ ] Improvement

# Description
Ack, Open, Close buttons don’t work for alarms.

### PR DESCRIPTION
**Pull request by maple10001**
# Type of change? 

- [x] Bug fix
- [ ] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
- fix #608 
...

**Checklist:**

- [x] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/components/spinner/spinner.scss [hunk 1]
```diff
@@ -1,355 +1,365 @@
...
+		  text-align: left;
+		  opacity: 0;
+		  transform-origin: 0 0;
+		  -webkit-transform-origin: 0 0;
+		  animation-iteration-count: infinite;
+		  animation-duration: 4000ms;
+		}
+
+		.dot:nth-child(1) {
+			transform:  rotate(225deg) translate(-50%, -50%);
```
Comment (swetha571) [inline src/components/spinner/spinner.scss]: Align the  "transform:  rotate(215deg) translate(-50%, -50%);" in whole file.

File: src/components/spinner/spinner.scss [hunk 2]
```diff
@@ -1,355 +1,365 @@
...
+        transform:rotate($deg1) translate(-50%, -50%);
+        animation-timing-function:ease-out;
+        opacity: 0;
+      }
+
+      1% { opacity: 1; }
+
+      6%{
+        transform:rotate($deg2) translate(-50%, -50%);
+        animation-timing-function:linear;
```
Comment (swetha571) [inline src/components/spinner/spinner.scss]: Add space after ":" operator.

File: src/components/pages/maintenance/maintenance.js [hunk 3]
```diff
@@ -119,6 +122,12 @@ class MaintenancePage extends Component {
         const ruleId = updatedAlarms[0].Rule.Id;
         this.props.actions.updateAlarmsStatus({ alarmIds, status, ruleId });
       }
+      setTimeout(() => {
+        this.setState({ showIndicator: false });
```
Comment (spryor) [inline src/components/pages/maintenance/maintenance.js]: Why hide the indicator on a timeout?

File: src/components/spinner/spinner.scss [hunk 4]
```diff
@@ -1,355 +1,365 @@
...
-    transform: rotate(945deg) translate(-50%, -50%);
-    animation-timing-function: ease-out;
-    opacity: 0;
-  }
-  100% {
-    transform: rotate(945deg) translate(-50%, -50%);
-    opacity: 0;
-  }
+	display: block;
+	overflow: hidden;
```
Comment (spryor) [inline src/components/spinner/spinner.scss]: This should be two spaces per tab.

File: src/components/pages/maintenance/maintenance.js (hunk 3 above)
Comment (maple10001) [reply src/components/pages/maintenance/maintenance.js]: design requirement to show at least one fully loading animation

File: src/components/pages/maintenance/maintenance.js [hunk 5]
```diff
@@ -110,6 +112,7 @@ class MaintenancePage extends Component {
 
   updateAlarm = (Status) => {
     const { selectedAlarms } = this.state;
+    this.setState({ showIndicator: true });
     Promise.all(
```
Comment (spryor) [inline src/components/pages/maintenance/maintenance.js]:
```
Observable.from(selectedAlarms)
    .do(_ => this.setState({ showIndicator: true }))
    .map(alarm => ({...alarm, Status})))
    .flatMap(ApiService.updateAlarmsStatus)
    .reduce((acc, curr) => [...acc, curr], [])
    .zip(Observable.of(undefined).delay(2000), updatedAlarms => updatedAlarms)
   .subscribe(updatedAlarms => { ...// handle success })
```

By using the zip, of the calls are faster than 2 seconds that is max time the spinner will appear. If the calls take longer than that, the spinner will appear for the length of the call.

**Pull request merged**